h3/src/server.rs

```diff
@@ -820,6 +820,20 @@ pin_project! {
         }
 }
 
+impl<'a, C, B> ReadDatagram<'a, C, B>
+where
+    C: quic::Connection<B>,
+    B: Buf,
+{
+    #[allow(missing_docs)]
+    pub fn new(conn: &'a mut Connection<C, B>) -> Self {
```

Comment:
Thanks for the PR.
This should be parte of the feature `i-implement-a-third-party-backend-and-opt-into-breaking-changes` because it is not part of the http3 spec.

In general I am not sure if just adding `new` functions is the right way to go. 
The examples you mentions do not break the WebTransport RFC and sound reasonable to me. In my opinion this should be supported by our `WebTransportSession`. So the right step would be to improve our API.
What do you think?

Reply:
`Request<()>` only be used in `fn validate_wt_connect(request: &Request<()>) -> bool`, is it possible move this validate outside and remove argument `request: Request<()>` from function accept?

Reply:
Would it help to make `WebTransport::accept(...)` generic like `WebTransport::accept<T>(request: Request<T>,...)`?

Reply:
Unfortunately, I use `CustomRequest` here, not just `Request<Custom>`. 
Another issue is that even if `Request<()>` is needed, it seems that ownership is not required here, and `&Request<()>` will suffice.

Reply:
Any new ideas or plans on this?